src/DefifaGovernor.sol

```diff
@@ -140,12 +157,71 @@ contract DefifaGovernor is
         return super.state(proposalId);
     }
 
+    /**
+     * @dev See {IGovernor-castVoteWithReason}.
+     * reverting to avoid passsing of duplicate tier id's in params
+     */
+    function castVoteWithReason(
+        uint256 proposalId,
+        uint8 support,
+        string calldata reason
+    ) public pure override returns (uint256) {
+        revert("use castVote");
+    }
+
+    /**
+     * @dev See {IGovernor-castVoteWithReasonAndParams}.
+     * reverting to avoid passsing of duplicate tier id's in params
+     */
+    function castVoteWithReasonAndParams(
+        uint256 proposalId,
+        uint8 support,
+        string calldata reason,
+        bytes memory params
+    ) public pure override returns (uint256) {
+        revert("use castVote");
+    }
+
+    /**
+     * @dev See {IGovernor-castVoteBySig}.
+     * reverting to avoid passsing of duplicate tier id's in params
+     */
+    function castVoteBySig(
+        uint256 proposalId,
+        uint8 support,
+        uint8 v,
+        bytes32 r,
+        bytes32 s
+    ) public pure override returns (uint256) {
+        revert("use castVote");
+    }
+
+
+    /**
+     * @dev See {IGovernor-castVoteWithReasonAndParamsBySig}.
+     * reverting to avoid passsing of duplicate tier id's in params
+     */
+    function castVoteWithReasonAndParamsBySig(
+        uint256 proposalId,
+        uint8 support,
+        string calldata reason,
+        bytes memory params,
+        uint8 v,
+        bytes32 r,
+        bytes32 s
+    ) public pure override returns (uint256) {
+        revert("use castVote");
+    }
+
     function propose(
         address[] memory targets,
         uint256[] memory values,
         bytes[] memory calldatas,
         string memory description
     ) public override(Governor) returns (uint256) {
+        if (block.timestamp <= proposalCreationThreshold) {
+            revert PROPOSAL_CREATION_THRESHOLD_NOT_REACHED_YET();
```

Comment:
OZ Governor has a method we can override for this functionality [`votingDelay`](https://docs.openzeppelin.com/contracts/4.x/api/governance#IGovernor-votingDelay--)

Reply:
> What do you think of doing it like this #5?

yh I thought of the same thing initially but I thought this might be expensive so instead I thought better to have only 1 callable `castVote` as i mentioned above but like you said we can't specify the tiers then will do a gas comparison just in case to be sure

Reply:
this is something else than the voting delay this wen the proposal creation can start

Reply:
voting delay is probably better in terms of ux maybe doesn't make sense to have both proposal and voting delay but maybe better to use the oz feature